dev/tools/examples_smoke_test.dart

```diff
@@ -0,0 +1,254 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This test builds an integration test from the list of samples in the
+// examples/api/lib directory, and then runs it. The tests are just smoke tests,
+// designed to start up each example and run it for a couple of frames to make
+// sure it doesn't throw an exception or fail to compile.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io' show stdout, stderr, exitCode, Process, ProcessException;
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+
+const bool kIsWeb = identical(0, 0.0);
+FileSystem filesystem = const LocalFileSystem();
+ProcessManager processManager = const LocalProcessManager();
+Platform platform = const LocalPlatform();
+
+FutureOr<dynamic> main() async {
+  if (!platform.isLinux && !platform.isWindows && !platform.isMacOS) {
+    stderr.writeln('Example smoke tests are only designed to run on desktop platforms');
+    exitCode = 4;
```

Comment:
TIL about `exitCode`. Is this equivalent to `exit(4);`?

Reply:
Yes, except that it allows for a clean shutdown.  If you call exit(4), then if the main is wrapped (e.g. in an integration test) then the test shutdown doesn't occur cleanly (it just exits the app immediately), and not all results are reported.